### PR TITLE
goread: init at 1.6.5

### LINKED ISSUE #326820: Package request: goread
**Project description**

goread is an RSS/Atom feed reader for your terminal. It allows you to categorize and follow feeds and read articles right in the terminal.

**Metadata**

* homepage URL: https://github.com/TypicalAM/goread
* source URL: https://github.com/TypicalAM/goread
* license: gpl3
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by schnow265**
## Description of changes

https://github.com/TypicalAM/goread

Terminal RSS/Atom feed reader made using golang and [bubbletea](https://github.com/charmbracelet/bubbletea).

Closes https://github.com/NixOS/nixpkgs/issues/326820

Continued from https://github.com/NixOS/nixpkgs/pull/315903

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5364,6 +5364,8 @@ with pkgs;
 
   gohai = callPackage ../tools/system/gohai { };
 
+  goread = callPackage ../by-name/go/goread { };
+
```
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:5368]:
```suggestion
```
unneeded with by-name

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution! As per guidelines, please rebase/squash into two commits:

* `maintainers: add schnow265`
* `goread: init at 1.6.5`

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -18050,6 +18050,12 @@
     githubId = 70169773;
     name = "Tarun Chinmai Sekar";
   };
+  schnow265 = {
```
Comment (eclairevoyant) [inline maintainers/maintainer-list.nix:18053]: this should go between `schneefux` and `schnusch` to be alphabetically sorted

**Comment by schnow265** [comment]
ok, will force-push and remove my commits and redo them.

File: pkgs/by-name/go/goread/package.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  vendorHash = "sha256-/kxEnw8l9S7WNMcPh1x7xqiQ3L61DSn6DCIvJlyrip0=";
+
+  meta = {
+    description = "Beautiful program to read your RSS/Atom feeds right in the terminal";
+    homepage = "https://github.com/TypicalAM/goread";
+    changelog = "https://github.com/TypicalAM/goread/releases/tag/v${version}";
+    license = lib.licenses.gpl3Plus;
+    mainProgram = "goread";
```
Comment (eclairevoyant) [inline pkgs/by-name/go/goread/package.nix:21]: I assume you plan to maintain this?

```suggestion
    mainProgram = "goread";
    maintainers = with lib.maintainers; [ schnow265 ];
```

File: pkgs/by-name/go/goread/package.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "TypicalAM";
+    repo = "goread";
+    rev = "v${version}";
+    hash = "sha256-SRVXcdgtRpWqvO28CnUcx40nFJnG+Hd94Ezgaj5xK6A=";
+  };
+
+  vendorHash = "sha256-/kxEnw8l9S7WNMcPh1x7xqiQ3L61DSn6DCIvJlyrip0=";
+
```
Comment (eclairevoyant) [inline pkgs/by-name/go/goread/package.nix:15]: https://github.com/TypicalAM/goread/blob/17d8570aa804a7ac483a2e0718dc6172aa4abae1/README.md?plain=1#L117

```suggestion

  env.TEST_OFFLINE_ONLY = 1;

```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Couple other points of feedback, other than that, tested and it seems fine

**Review by eclairevoyant — APPROVED** [review]

**Pull request merged**